src/Symfony/Component/Security/Core/Authorization/Voter/ExplainedVoterInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+interface ExplainedVoterInterface
```

Comment:
the additional API is not really needed right? it could be a detail of the current abstract Voter class

we ask the voter for a decision, im not sure in what case one would know to call either grant()/abstain()/deny() upfront

Reply:
> we ask the voter for a decision

There is just a little confusion : we ask the voter for 1 **vote** and the **decision** is taken from all the votes (and according to a strategy).